__tests__/hooks/pager.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import { User } from 'src/contrib/aidbox';
+import { usePager } from '../../hooks/pager';
+import { useService } from '../../hooks/service';
+import { success } from '../../libs/remoteData';
+import { getFHIRResources } from '../../services/fhir';
+
+jest.mock('../../services/fhir', () => ({ getFHIRResources: jest.fn() }));
+jest.mock('../../hooks/service', () => ({ useService: jest.fn() }));
+
+describe('Hook `usePager`', () => {
+    const resourceType = 'User';
+    const resourcesOnPage = 2;
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('property `hasNext`', () => {
+        test("returns false when there's no next page", () => {
+            const data = success({
+                id: 'fakeID',
+                resourceType: 'type',
+            });
+
+            (<jest.Mock>useService).mockImplementation(() => [data]);
+
+            const { result } = renderHook(() => usePager<User>(resourceType, resourcesOnPage));
+            const [remoteData, { hasNext }] = result.current;
+
+            expect(hasNext).toBeFalsy();
+            expect(remoteData).toEqual(data);
+        });
+
+        test("returns true when there's next page", () => {
+            const data = success({
+                id: 'fakeID',
+                resourceType: 'type',
+                link: [
+                    {
+                        resourceType: 'Bundle',
+                        relation: 'next',
+                        uri: 'uri',
+                    },
+                ],
+            });
+
+            (<jest.Mock>useService).mockImplementation(() => [data]);
+
+            const { result } = renderHook(() => usePager<User>(resourceType, resourcesOnPage));
+            const [remoteData, { hasNext }] = result.current;
+
+            expect(hasNext).toBeTruthy();
+            expect(remoteData).toEqual(data);
+        });
+    });
+
+    test('method `loadNext`', async () => {
+        const data = success({
+            id: 'fakeID',
+            resourceType: 'type',
+        });
+
+        (<jest.Mock>useService).mockImplementation(() => [data]);
+
+        const { result } = renderHook(() => usePager<User>(resourceType, resourcesOnPage));
+        const { loadNext } = result.current[1];
```

Comment:
Too hard to read.
Let's rewrite:
```
const [remoteData, { loadNext} ] = result.current;

```

Reply:
At https://github.com/beda-software/aidbox-react/blob/3805ee3c10fccdbfd8f3411629cb66dc92724a42/__tests__/hooks/crud.spec.ts#L33-L37
you use destructuring



It makes sense to extract values in the same way in all cases.